tests/unit/Endpoints/Webhooks/InfoHandlerTest.php

```diff
@@ -59,69 +98,108 @@ public function testHandleOk()
+        $taskProcessorStatus = $systemStatus['task_processor'];
+        $failedCompatibilityChecks = $systemStatus['failed_compatibility_checks'];
 
-        $response = $this->handleRequest($rest);
-        $data = $response->get_data();
-        $extra = $data['extra'];
-
-        $this->assertEquals(
-            StoreKeeperOptions::SYNC_MODE_FULL_SYNC,
-            $extra['sync_mode'],
-            'Incorrect sync mode'
-        );
+        // Assert system status
 
-        /*
-         * Check sync stats
-         */
-        $this->assertEquals(
-            InfoHandler::getLastSyncRunDate(),
-            $extra['last_sync_run_date'],
-            'Incorrect last_sync_run_date'
-        );
-        $this->assertEquals(
-            InfoHandler::getLastHookDate(),
-            $extra['last_hook_date'],
-            'Incorrect last_hook_date'
-        );
+        // Assert order system status
+        $this->assertEquals($expectedIdsNotSynchronized, $orderSystemStatus['ids_not_synchronized'], 'Not synchronized IDs should match with extras');
+        $this->assertEquals($expectedFailedOrderIds, $orderSystemStatus['ids_with_failed_tasks'], 'Failed order IDs should match with extras');
+        $this->assertEquals($expectedLastOrderDate, $orderSystemStatus['last_date']);
         $this->assertEquals(
-            TaskModel::countTasks(),
-            $extra['task_quantity'],
-            'Incorrect task_quantity'
+            \DateTime::createFromFormat(
+                'Y-m-d H:i:s',
+                TaskModel::getLatestSuccessfulSynchronizedDateForType(TaskHandler::ORDERS_EXPORT)
+            )->format(DATE_RFC2822),
+            $orderSystemStatus['last_synchronized_date'],
+            'Last successful synchronized order ID should match with extras'
         );
         $this->assertEquals(
-            WebhookLogModel::count(),
-            $extra['hook_quantity'],
-            'Incorrect hook_quantity'
-        );
-        $this->assertEquals(
-            InfoHandler::getLastWebhookLogId(),
-            $extra['last_hook_id'],
-            'Incorrect last_hook_id'
+            $expectedOldestDateNotSynchronized,
+            $orderSystemStatus['oldest_date_not_synchronized'],
+            'Oldest order unsynchronized date should match with extras'
         );
+
+        // Assert task processor
         $this->assertEquals(
-            InfoHandler::getLastTaskId(),
-            $extra['last_task_id'],
-            'Incorrect last_task_id'
+            TaskModel::count(['status = :status'], ['status' => TaskHandler::STATUS_NEW]),
+            $taskProcessorStatus['in_queue_quantity'],
+            'Task in queue should match with extras'
         );
+
         $this->assertEquals(
-            TaskModel::countFailedTasks(),
-            $extra['task_failed_quantity'],
-            'Incorrect task_failed_quantity'
+            \DateTime::createFromFormat('Y-m-d H:i:s', TaskModel::getLastProcessTaskDate())->format(DATE_RFC2822),
```

Comment:
that uses internal code which is executed as $expected value, so not a really good test

Reply:
Modified to get expected date from expected order.